### ISSUE TITLE
[BUG] mydumper --daemon mode leaving DB in Global Read Lock (limbo)

### ISSUE THREAD

**Issue by madhurbhaiya** [mydumper/mydumper#1009]
**Describe the bug**
I used mydumper in ```--daemon``` mode. In the second run (after roughly 1 hour interval of first run's successful completion), it acquires ```FLUSH TABLES WITH READ LOCK``` and then gets stuck leaving complete database under ```GLOBAL READ LOCK```.

This script ran during midnight, and due to this error, the complete application went down because of the read lock.

**To Reproduce**
Command executed:
```
mydumper --host=127.0.0.1 --port=3306 --user=dbuser --ask-password --threads=16 --compress --compress-protocol --build-empty-files --routines --events --trx-consistency-only --use-savepoints --checksum-all --rows=50000 --verbose=3 --daemon --logfile=mydumper.log --outputdir=dbname-bkup --database=dbname
```

What mydumper version has been used?

**mydumper0.13.1-2, built against MySQL 8.0.30-22 with SSL support with GZIP**

**Expected behavior**
1. It should have run successfully - by taking the incremental backup and then released the read lock, instead of getting stuck in limbo. But in case, it gets stuck somewhere, it should break the execution and release the acquired locks. Log the error and exit gracefully.

2. I would prefer a flag to specify the times that the mydumper should execute during daemon mode. Basically, my understanding of the daemon mode was that it runs the mydumper in "background" and finishes its execution once the dump is done successfully. I expected it to behave like ```nohup ... &``` option. An alternative would be to provide a new flag ```--background``` which simulates ```nohup ... &``` behaviour.

**Log**
If applicable, add the --verbose 3 / -v 3 and --logfile / -L <filename> to the execution of the command and upload the file.

```
2023-01-26 03:34:38 [INFO] - Thread 5: shutting down
2023-01-26 03:34:38 [INFO] - Thread 2: shutting down
2023-01-26 03:34:38 [INFO] - Thread 13: shutting down
2023-01-26 03:34:38 [INFO] - Thread 9: shutting down
2023-01-26 03:34:38 [INFO] - Thread 10: shutting down
2023-01-26 03:34:38 [INFO] - Thread 4: shutting down
2023-01-26 03:34:38 [INFO] - Thread 6: shutting down
2023-01-26 03:34:38 [INFO] - Thread 7: shutting down
2023-01-26 03:34:38 [INFO] - Thread 15: shutting down
2023-01-26 03:34:38 [INFO] - Thread 1: shutting down
2023-01-26 03:34:38 [INFO] - Thread 14: shutting down
2023-01-26 03:34:38 [INFO] - Thread 12: shutting down
2023-01-26 03:34:38 [INFO] - Thread 16: shutting down
2023-01-26 03:34:38 [INFO] - Thread 8: shutting down
2023-01-26 03:34:38 [INFO] - Thread 11: Table activity_log completed 
2023-01-26 03:34:38 [INFO] - Thread 11: shutting down
2023-01-26 03:34:38 [INFO] - Thread 3: shutting down
2023-01-26 03:34:38 [INFO] - Releasing DDL lock
2023-01-26 03:34:38 [INFO] - Queue count: 0 0 0 0 0
2023-01-26 03:34:38 [INFO] - Main connection closed
2023-01-26 03:34:38 [INFO] - Finished dump at: 2023-01-26 03:34:38
2023-01-26 04:34:38 [INFO] - Server version reported as: 8.0.30-22
2023-01-26 04:34:38 [INFO] - Connected to a MySQL server
2023-01-26 04:34:38 [INFO] - Sending Flush Table
2023-01-26 04:34:38 [INFO] - Acquiring FTWRL
2023-01-26 04:34:38 [INFO] - Acquiring DDL lock
2023-01-26 04:34:38 [INFO] - Started dump at: 2023-01-26 04:34:38
```

**Environment (please complete the following information):**
 - OS version: **Ubuntu 22.04.1 LTS**
 - MyDumper version: **mydumper0.13.1-2, built against MySQL 8.0.30-22 with SSL support with GZIP**

**Comment by davidducos** [comment]
@madhurbhaiya, please, do not use --daemon until this is fixed. 

We are not cleaning and creating correctly each execution.

**Comment by viswaprsnl** [comment]
@davidducos is this still a valid problem in 0.15 version ,global DDL lock .lets say if i start an export , during my entire export time (lets say 10 hr) . Data base is locked ?

**Comment by davidducos** [comment]
@viswaprsnl it should not be a problem anymore. If your database is being locked by mydumper, please describe your problem in a new issue.
